bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -125,7 +125,10 @@ impl Connection {
 
     /// Reports the remaining nanoseconds before the connection may be safely closed.
     ///
-    /// If [`shutdown`] is called before this method reports "0", then an error will occur.
+    /// If [`poll_shutdown`] is called before this method reports "0", then an error will occur.
+    ///
+    /// This method is expected to succeed, but could fail if the underlying C call encounters errors.
+    /// If it fails, a graceful two-way shutdown of the connection will not be possible.
```

Comment:
Is the only way this can fail is if `poll_shutdown` is called before? What does a not-graceful shutdown look like (leaked resources, abrupt connection close with no alerts)? In normal/correct operation should this fail?